iphone/Maps/UI/Search/TableView/MWMSearchCommonCell.mm

```diff
@@ -80,19 +80,40 @@ - (void)config:(search::Result const &)result
     }
   }
 
-  bool popularityHasHigherPriority = PopularityHasHigherPriority(lastLocation, distanceInMeters);
-  bool showClosed = result.IsOpenNow() == osm::No;
   bool showPopular = result.GetRankingInfo().m_popularity > 0;
-
-  if (showClosed && showPopular)
+  self.popularView.hidden = !showPopular;
+  self.openLabel.hidden = true;
```

Comment:
Move it below into the final (missing) default switch clause, to avoid unnecessary hiding/showing.